### PR TITLE
doc: layout-config-reference: Document supported file types

### PR DESCRIPTION
**Pull request by landerweit-phytec**
Document behavior for different file types.

Signed-off-by: Leonard Anderweit <l.anderweit@phytec.de>

### CODE REVIEW

**Review by mschwan-phytec — CHANGES_REQUESTED** [review]

File: doc/layout-config-reference.rst [hunk 1]
```diff
@@ -127,3 +127,17 @@ optional checksum can be provided with ``md5sum`` and/or ``sha256sum``.
...
    against the provided file before writing to the target partition or volume.
+
+Supported File Types
+....................
+
+The provided input files are copied to the filesystem of the corresponding
+partition. If files are of one of the supported special file types, they are
+treated as following.
+
+``tar`` archives and ``tar.gz`` compressed archives are extracted into the
```
[the remark this answers is not in the document]
Comment (landerweit-phytec) [reply doc/layout-config-reference.rst]: Fixed with `tar.*` to include all compressed archives.

File: doc/layout-config-reference.rst [hunk 2]
```diff
@@ -127,3 +127,17 @@ optional checksum can be provided with ``md5sum`` and/or ``sha256sum``.
...
+filesystem.
+
+``ext[234]`` raw filesystem files are written directly to the partition. This
+overrides the existing filesystem. Additionally ext4 filesystems are resized to
```
[the remark this answers is not in the document]
Comment (landerweit-phytec) [reply doc/layout-config-reference.rst]: Noted this detail.

File: doc/layout-config-reference.rst [hunk 3]
```diff
@@ -127,3 +127,19 @@ optional checksum can be provided with ``md5sum`` and/or ``sha256sum``.
...
+The provided input files are copied to the filesystem of the corresponding
+partition. If files are of one of the supported special file types, they are
+treated as following.
+
+``tar`` or ``tar.*``
+   Archives and compressed archives are extracted into the filesystem.
+
+``ext[234]``
+   Raw filesystem files are written directly to the partition. This overrides
+   the existing filesystem, so the filesystem should be specified as
```
Comment (mschwan-phytec) [inline doc/layout-config-reference.rst:143]:
```diff
-the existing filesystem
+any existing filesystem
```
```diff
-, so the filesystem
+, so it
```

File: doc/layout-config-reference.rst [hunk 4]
```diff
@@ -127,3 +127,19 @@ optional checksum can be provided with ``md5sum`` and/or ``sha256sum``.
...
+   ``filesystem: null`` or not be specified at all. Additionally ext4
```
Comment (mschwan-phytec) [inline doc/layout-config-reference.rst:144]: s/ext4/ext/g

So it does not sound like only ext4 filesystems are resized afterwards.

**Review by mschwan-phytec — CHANGES_REQUESTED** [review]

**Comment by landerweit-phytec** [comment]
Thanks for the suggestions. Updated the commit accordingly.

**Review by mschwan-phytec — APPROVED** [review]

**Pull request merged**